apps/api/src/withdrawal/withdrawal.service.ts

```diff
@@ -124,11 +118,25 @@ export class WithdrawalService {
     return result
   }
 
-  // Functionality will be reworked soon
   async remove(id: string): Promise<Withdrawal | null> {
-    throw new ForbiddenException()
-    const result = await this.prisma.withdrawal.delete({ where: { id: id } })
-    if (!result) throw new NotFoundException('Not found')
+    const result = await this.prisma.withdrawal
+      .delete({
+        where: { id: id },
+      })
+      .catch(() => {
+        throw new BadRequestException("Withdrawal record couldn't be deleted")
+      })
+
```

Comment:
I think it is better to check the status before updating, instead of incrementing with 0...